public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -280,16 +285,66 @@ export class LokiDatasource
 
     if (fixedRequest.liveStreaming) {
       return this.runLiveQueryThroughBackend(fixedRequest);
-    } else {
-      const startTime = new Date();
-      return super.query(fixedRequest).pipe(
-        // in case of an empty query, this is somehow run twice. `share()` is no workaround here as the observable is generated from `of()`.
-        map((response) =>
-          transformBackendResult(response, fixedRequest.targets, this.instanceSettings.jsonData.derivedFields ?? [])
-        ),
-        tap((response) => trackQuery(response, fixedRequest, startTime))
-      );
     }
```

Comment:
the `loki/datasource.ts` file is already very long, and considering this is a feature-flag only feature, may be a good idea to implement it in a separate file, and just call it from here.

Reply:
that should also allow us to be more confident that these changes are not "active" without the feature-flag.

Reply:
Totally. Done.